hail/python/hailtop/hailctl/config/cli.py

```diff
@@ -85,7 +70,47 @@ def set(parameter: str, value: str):
 
 
 @app.command()
-def unset(parameter: str):
+def set(parameter: Ann[ConfigVariable, Arg(help="Configuration variable to set", autocompletion=complete_config_variable)], value: str):
+    if parameter not in config_variables():
+        print(f"Error: unknown parameter {parameter!r}", file=sys.stderr)
+        sys.exit(1)
+
+    section, key, path = get_section_key_path(parameter.value)
+
+    validations = {var: var_info.validation for var, var_info in config_variables().items()}
+
+    validation_func, msg = validations.get(parameter, (lambda _: True, ''))  # type: ignore
```

Comment:
We should be able to assert now that `parameter` is in the map. We can make a test that asserts every `ConfigVariable` is in `config_variables` (the `ConfigVariable` class itself is iterable so you can loop over all the variants of the enum).